base/settings/R/pft_site_linker.R

```diff
@@ -0,0 +1,66 @@
+#'
+#' @title site.pft.linkage
+#' 
+#' @param pecan.xml.address pecan settings list.
+#' @param site.pft.links dataframe. Your look up table should have two columns of site and pft with site ids under site column and pft names under pft column.
+#'
+#' 
+#' @description This function creates the required tags inside pecan.xml to link sites with pfts given a look up table. If the required tags are already defined in the pecan xml then they will be updated.
+#' 
+#' @return NONE
+#' @export site.pft.linkage
+#' 
+#' @examples
+#'\dontrun{
+#' #setting up the Look up tables
+#' site.pft.links <-tribble(
+#'  ~site, ~pft,
+#'  "1000025731", "temperate.broadleaf.deciduous1",
+#'  "1000025731", "temperate.broadleaf.deciduous11",
+#'  "1000000048", "temperate.broadleaf.deciduous2",
+#'  "772", "temperate.broadleaf.deciduous3",
+#'  "763", "temperate.broadleaf.deciduous4"
+#' )
+#'  
+#' # sending a single setting xml to the function
+#' site.pft.linkage(settings,site.pft.links)
+#' # sending a multi- setting xml file to the function
+#' site.pft.linkage(settings,site.pft.links)
+#'}
+site.pft.linkage <- function(settings, site.pft.links){
+  
+  # checking the LUT 
+  if (is.null(site.pft.links) | ncol(site.pft.links) !=2) PEcAn.logger::logger.severe('Your look up table should have two columns of site and pft with site ids under site column and pft names under pft column.')
+
+  # if it's not a multisetting put it still in a list
+  if(!is.MultiSettings(settings)) settings<- list(settings) 
+  
+  #for each site in this setting
+  new.mset <- settings%>%
+    purrr::map(function(site.setting){
+      
+      site.pft <- NULL
+      site.id <- (site.setting[['run']])$site$id
+      #if no site id was found
+      if (is.null(site.id)) PEcAn.logger::logger.severe('Your site needs to have a site id.')
```

Comment:
* do you really want to stop a full multiSettings run if one site is missing a site ID?
* Would be nice to have a more informative error message -- if you're running 1000's of sites and 1 is missing and ID, this would be a pretty uninformative message.